tests/posix/common/src/pthread.c

```diff
@@ -539,6 +539,35 @@ ZTEST(pthread, test_pthread_testcancel)
 	zassert_false(testcancel_failed);
 }
 
+static void *test_pthread_setschedprio_fn(void *arg)
+{
+	pthread_t self;
+	int prio = 0;
+	int policy;
+	struct sched_param param;
+
+	param.sched_priority = K_HIGHEST_APPLICATION_THREAD_PRIO;
+
+	self = pthread_self();
```

Comment:
Just some nits. I think we can drop the highest application priority part.
```suggestion
	int policy;
	int prio = 0;
	struct sched_param param;
	pthread_t self = pthread_self();
```